src/engine/ruleset/integrations/windows/decoders/windows-sysmon.yml

```diff
@@ -16,21 +16,175 @@ metadata:
 
+definitions:
+  idEvent_map:
+    "1":
+      category:
+        - process
+      type:
+        - start
+    "2":
+      category:
+        - file
+      type:
+        - change
+    "3":
+      category:
+        - network
+      type:
+        - start
+        - connection
+        - protocol
+    "4":
+      category:
+        - process
+      type:
+        - change
+    "5":
+      category:
+        - process
+      type:
+        - end
+    "6":
+      category:
+        - driver
+      type:
+        - start
+    "7":
+      category:
+        - process
+      type:
+        - change
+    "8":
+      category:
+        - process
+      type:
+        - change
+    "9":
+      category:
+        - process
+      type:
+        - access
+    "10":
+      category:
+        - process
+      type:
+        - access
+    "11":
+      category:
+        - file
+      type:
+        - creation
+    "12":
+      category:
+        - configuration
+        - registry
+      type:
+        - change
+    "13":
+      category:
+        - configuration
+        - registry
+      type:
+        - change
+    "14":
+      category:
+        - configuration
+        - registry
+      type:
+        - change
+    "15":
+      category:
+        - file
+      type:
+        - access
+    "16":
+      category:
+        - configuration
+      type:
+        - change
+    "17":
+      category:
+        - file
+      type:
+        - creation
+    "18":
+      category:
+        - file
+      type:
+        - access
+    "19":
+      category:
+        - process
+      type:
+        - creation
+    "20":
+      category:
+        - process
+      type:
+        - creation
+    "21":
+      category:
+        - process
+      type:
+        - access
+    "22":
+      category:
+        - network
+      type:
+        - connection
+        - protocol
+        - info
+    "23":
+      category:
+        - file
+      type:
+        - deletion
+    "24":
+      type:
+        - change
+    "25":
+      category:
+        - process
+      type:
+        - change
+    "26":
+      category:
+        - file
+      type:
+        - deletion
+    "27":
+      category:
+        - file
+      type:
+        - creation
+        - denied
+    "28":
+      category:
+        - file
+      type:
+        - deletion
+        - denied
+    "255":
+      category:
+        - process
+      type:
+        - error
+
 check:
   - windows.System.Provider.@Name: Microsoft-Windows-Sysmon
 
-
 normalize:
  - map:
     - wazuh.decoders: array_append(windows-sysmon)
     - event.dataset: sysmon
     - event.kind: event
+    - event: merge_value($idEvent_map, $event.code)
 
     # Add categorization fields
     - rule.name: $windows.EventData.RuleName
     - event.hash: $windows.EventData.Hash
-    - _hashes: parse_key_value($windows.EventData.Hashes, =, ',', '\'', '\'')
-    - related.hash: $_hashes
+    - related.hash: $windows.EventData.Hashes
```

Comment:
is this an array of strings?

Reply:
yes

Reply:
no is not, is a string:
```xml
<Data Name="Hashes">MD5=55BFEBE68443611320A8A70C7DFD5D56,SHA256=5809ABFE5219B046AF113536279F8B16F9D6A5A3E63417BAE6E1202A3E958C33,IMPHASH=00000000000000000000000000000000</Data>
```

It can also be seen at its exit:
![image](https://github.com/wazuh/wazuh/assets/1891958/459e86c1-8029-4d88-8c02-0fa1499ff9b6)


Check:

https://github.com/elastic/ecs/blob/912da8f2173b75116d9d3bff6ed7472b10f871bb/generated/csv/fields.csv#L1010